Update and refactor CoC policy and processes

tobie opened the pull request:
- Clearly split all CoC related-documents into 4 buckets:
    1. The **Code of Conduct** itself, which is also the source of truth for the makeup of the CoC Team, how to contact the CoC Team, and how to report an incident, escalate it, and appeal a decision.
    2. The **Code of Conduct Policy**, which defines the higher level aspects of our code of conduct implementation (the _what_).
    3. The **CoC Team Charter** which is the mean by which the CPC delegates responsibilities to the CoC Team.
    4. Various related **processes** which describe _how_ the policy is implemented.
- Replace the Code of Conduct Panel with a Code of Conduct Team instead.
- Standardize on the language used in the Contributor Covenant and in the CNCF and use it consistently throughout.
- Clarify the escalation and appeal policies.
- Be explicit about the projects's freedom to implement moderation policies
- Create a placeholder for all of the processes related to the makeup and structure of the CoC Team.
- Add an indication at the top of each related file pointing people who just want to report an incident to the right place.
- Clean up a number of smaller issues.
- Clarify the requirements for projects who opt-in to manage incident reports themselves. Make the policy generic so that it applies both to those projects and to the CoC. Partially addresses #786.

### Still to do:
- [x] Decide whether it is better to separate report@, coc-escalation@, and  coc-appeal@ or just use report@ for all the things. Decided to use just one email: [coc@openjsf.org](mailto:coc@openjsf.org)
    - [x] Add or remove email lists accordingly
    - [x] Update the CoC accordingly.
- [x] Add the CoC Team Charter (and in particular clarify the makeup and selection process of the CoC Team) (@PaulaPaul).
- [x] Should the privacy policy also protect the accused person, and if so, what are the necessary exceptions? 
- [x] Reach out to projects who would be affected by these changes (tracking in https://github.com/openjs-foundation/cross-project-council/issues/1255).
- [x]  Address concerns raised by @bnb:
  - [x] There's no clear definition of conflict of interest. Agreement that we need a conflict of interest policy, but can we land this PR without it? ([Link to comment](#discussion_r1446417120))
  - [x] What constitutes report mishandling needs to be defined more clearly. ([Link to comment](#discussion_r1446407933))
  - [x] The escalation and appeal policy described in this PR allow the CoC Team to reverse decisions made at the project level in certain fairly scoped situations. There isn't consensus about this. ([Link to comment](#discussion_r1446422772))
  - [x] The current policy delegates defining case handling processes to the CoC Team. There isn't consensus about this. ([Link to comment](#discussion_r1446431534))
  - [x] There isn't consensus that having a CoC Chair is useful. ([Link to comment](#discussion_r1446436310))
  - [x] There isn't consensus about eligibility criteria. ([Link to comment](#discussion_r1446439889))
  
### This is now ready to land pending:
  - [x] @bnb to confirm they're no longer blocking (commenting didn't remove the block) (marked as done as we can work around the technical issue)
  - [x] @eemeli to confirm he's happy with the provisions around confidentiality of accused individuals
  - [x] agreement to remove obsolete incident management process and reference it by a link to versioned content

mhdawson commented on conduct/FOUNDATION_CODE_OF_CONDUCT_REQUIREMENTS.md:
```diff
@@ -1,32 +1,66 @@
-projects within it. This document defines the requirements
-for a CoC for the Foundation and the projects which are part
-of the foundation.
+The OpenJS Foundation (OpenJSF) delegatea to the Cross Project Council (CPC), via its [charter][], the responsability of chosing, maintaining, and enforcing a Code of Conduct (CoC) for all of its projects, collab spaces, and for any related activities.
```
```suggestion
The OpenJS Foundation (OpenJSF) delegates to the Cross Project Council (CPC), via its [charter][], the responsability of chosing, maintaining, and enforcing a Code of Conduct (CoC) for all of its projects, collab spaces, and for any related activities.
```

ljharb commented on conduct/FOUNDATION_CODE_OF_CONDUCT_REQUIREMENTS.md:
```diff
@@ -1,32 +1,66 @@
+
+Projects may opt-in to enforce the CoC themselves provided they meet the requirements [described below][delegation to projects]. However, reportees, victims, and offenders may escalate an issue to the CoC Enforcement Team at any point in the process to avoid a conflict of interest and appeal the decision up to thirty (30) days after a response to the complaint was provided if they believe that th ...
+
+For spaces managed by the Foundation, complaints can be forwarded to report@openjsf.org and will
+be investigated by the Cross Project Council (CPC) or its delegates.
```
```suggestion
For spaces managed by the Foundation, complaints can be forwarded to report@openjsf.org and will be investigated by the Cross Project Council (CPC) or its delegates.
```

ljharb commented on conduct/FOUNDATION_CODE_OF_CONDUCT_REQUIREMENTS.md:
```diff
@@ -70,3 +104,18 @@ a yearly basis.
+
+## Requirements for projects who opt-in to manage enforcement of the CoC themselves.
+
+For spaces managed by member projects, the project must provide an email to which complaints
+can be forwarded and document that complaints will be investigated and will result in a
+response that is deemed appropriate for the circumstances. Each project is free to
+implement enforcement of the Code of Conduct using processes appropriate for the size
+and scope of the project including any day-to-day moderation of the project spaces. As part
+of these processes, however, the project must agree to:
```
```suggestion
For spaces managed by member projects, the project must provide an email to which complaints can be forwarded and document that complaints will be investigated and will result in a response that is deemed appropriate for the circumstances.
Each project is free to implement enforcement of the Code of Conduct using processes appropriate for the size and scope of the project including any day-to-day moderation of the project spaces.
As part of these processes, however, the project must agree to:
```

ljharb commented on CODE_OF_CONDUCT.md:
```diff
@@ -23,18 +14,25 @@ All recipients of reports commit to:
+## Code of Conduct Enforcement Team (CoCET)
 
-## Escalate an issue
+The Code of Conduct Enforcement Team (CoCET) is a foundation-wide team established to manage code of conduct incident reports across the whole foundation (except for [projects who have opted-in][] to manage reports directly), escalation of reports managed by projects, and appeals to decisions made in response to incidents. The CoCET is currently composed of:
```
where do/will we document who selects this team and related info?

mhdawson replied:
I was just typing in the same comment. It needs to be documented how the team members are selected.

ljharb replied:
it probably shouldn't be in the CoC directly - just somewhere linked from here :-)

tobie replied:
Yeah, I'm drafting this and figuring out the order of stuff as I go. It'll go in COC_POLICY and maybe a little in COC_PROCESS too. This is very drafty right now, tbh.

tobie replied:
That said, I do think that the members of the CoCET team should be listed in the COC (so reporters can reach out directly if they prefer) and that a line to explain what the CoCET is and referencing its full definition is thus warranted.

ljharb replied:
Totally agree; list current members inline, but link to full context/process/etc elsewhere.

mhdawson replied:
I agree they should be listed, with a link to doc on how they are selected and approved.

mcollina reviewed (approved):
lgtm

VoltrexKeyva commented on CODE_OF_CONDUCT.md:
```diff
@@ -23,23 +14,29 @@ All recipients of reports commit to:
     * **Webhint:** <support@webhint.io>
-2. For every other OpenJS Foundation project, please email <report@lists.openjsf.org>. The Cross Project Council (CPC) is responsible for managing these reports.
+    
+    In case of a conflict of interest or if the report is not handled appropriately, you may [escalate the report][escalation] to the [Code of Conduct Team (CoC Team)][CoC Team] by emailing <coc-escalation@lists.openjsf.org> or by contacting any if its members.
+2.  **To report any other incident** (for example: in an OpenJS Foundation project not listed above, in a Collab Space, in the Cross Project Council, or during an event), please email the [Code of Conduct Team (CoC Team)][CoC Team] at <report@lists.openjsf.org> or contact any if its members.
```
```suggestion
    In case of a conflict of interest or if the report is not handled appropriately, you may [escalate the report][escalation] to the [Code of Conduct Team (CoC Team)][CoC Team] by emailing <coc-escalation@lists.openjsf.org> or by contacting any of its members.
2.  **To report any other incident** (for example: in an OpenJS Foundation project not listed above, in a Collab Space, in the Cross Project Council, or during an event), please email the [Code of Conduct Team (CoC Team)][CoC Team] at <report@lists.openjsf.org> or contact any of its members.
```

VoltrexKeyva commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,99 @@
+
+# Code of Conduct Policy
+
+The OpenJS Foundation (OpenJSF) delegates to the Cross Project Council (CPC), via its [charter][], the responsability of chosing, maintaining, and enforcing a Code of Conduct (CoC) for all of its projects, collab spaces, and for any related activities.
```
```suggestion
The OpenJS Foundation (OpenJSF) delegates to the Cross Project Council (CPC), via its [charter][], the responsibility of choosing, maintaining, and enforcing a Code of Conduct (CoC) for all of its projects, collab spaces, and for any related activities.
```

VoltrexKeyva commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,99 @@
+
+A common code of conduct allows a common escalation and appeal process and allows better support of projects in their efforts to apply the code of conduct.
+
+All OpenJSF projects, collab spaces, and any activites related to the OpenJSF are operating under this code of conduct unless otherwise stated. (For example, events organized in partnership with other organizations might be operating under a different code of conduct.)
```
```suggestion
All OpenJSF projects, collab spaces, and any activities related to the OpenJSF are operating under this code of conduct unless otherwise stated. (For example, events organized in partnership with other organizations might be operating under a different code of conduct.)
```

VoltrexKeyva commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,99 @@
+
+All OpenJSF projects, collab spaces, and any activites related to the OpenJSF are operating under this code of conduct unless otherwise stated. (For example, events organized in partnership with other organizations might be operating under a different code of conduct.)
+
+Projects joining the foundation who already have a code of conduct will need to adopt the OpenJSF's CoC within three (3) months. Projects who don't already have a code of conduct operate under the OpenJSF's CoC as soon as they enter incubation.
```
```suggestion
Projects joining the foundation that already have a code of conduct will need to adopt the OpenJSF's CoC within three (3) months. Projects that don't already have a code of conduct operate under the OpenJSF's CoC as soon as they enter incubation.
```

VoltrexKeyva commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,99 @@
+1. A general email address at which incidents can be reported.
+2. The name and email address of each recipient of the general email address.
+
+Reporters may chose to report an incident by using the general email address provided or by contacting any of its recipient individually, either by using their provided email, reaching out in person, or by any other reasonable mean.
```
```suggestion
Reporters may choose to report an incident by using the general email address provided or by contacting any of its recipients individually, either by using their provided email, reaching out in person, or by any other reasonable means.
```

VoltrexKeyva commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,99 @@
+
+Reporters may chose to report an incident by using the general email address provided or by contacting any of its recipient individually, either by using their provided email, reaching out in person, or by any other reasonable mean.
+
+An acknowledgement of receipt should be promptly issued to incident reporters.
```
```suggestion
An acknowledgment of receipt should be promptly issued to incident reporters.
```
For consistency of using American English.

VoltrexKeyva commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,99 @@
+
+The OpenJSF makes the following commitments to reporters and victims of CoC incidents:
+
+1. All incident reports will be reviewed, investigated, and will result in a response that is deemed necessary and appropriate to the circumstances.
```
```suggestion
1. All incident reports will be reviewed and investigated and will result in a response that is deemed necessary and appropriate to the circumstances.
```

VoltrexKeyva commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,99 @@
+
+## Conflict of Interest
+
+Any person involved with the management of an incident report must immediatly notify the CoC Team in written and recuse themselves from the process if they have been involved in the incident itself or have any other conflict of interest with any of the involved parties (e.g. same employer).
```
```suggestion
Any person involved with the management of an incident report must immediately notify the CoC Team in writing and recuse themselves from the process if they have been involved in the incident itself or have any other conflict of interest with any of the involved parties (e.g. same employer).
```

VoltrexKeyva commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,99 @@
+
+The structure of the CoC Team and the process by which its members are chosen are described in the [CoC Team processes][].
+
+The current composition of the CoC Team is available in the [code of conduct][CoC] itself along with means to contact by email the whole CoC Team and each of it members individually.
```
```suggestion
The current composition of the CoC Team is available in the [code of conduct][CoC] itself along with means to contact by email the whole CoC Team and each of its members individually.
```

VoltrexKeyva commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,99 @@
+
+### Delegation to projects
+
+Projects may opt-in to enforce the code of conduct themselves provided they meet the following requirement:
```
```suggestion
Projects may opt-in to enforce the code of conduct themselves provided they meet the following requirements:
```

VoltrexKeyva commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,99 @@
+
+## Escalation
+
+The CoC Team is empowered to manage escalation requests of incident reports managed by projects which have opted-in to enforce the code of conduct themselves.
```
```suggestion
The CoC Team is empowered to manage escalation requests of incident reports managed by projects that have opted-in to enforce the code of conduct themselves.
```

VoltrexKeyva commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,99 @@
+
+If the initial report was already managed by the CoC Team, a different set of CoC Team members should be chosen to manage the appeal process.
+
+If the appeal follows an incident report managed by a project, the project is expected to participate to the appel process in good faith. 
```
```suggestion
If the appeal follows an incident report managed by a project, the project is expected to participate in the appeal process in good faith. 
```

VoltrexKeyva commented on conduct/COC_PROCESS_FOR_COC_TEAM.md:
```diff
@@ -0,0 +1,19 @@
+This document defines how the the OpenJS Foundation implements its code of conduct.
```
```suggestion
This document defines how the OpenJS Foundation implements its code of conduct.
```

VoltrexKeyva commented on conduct/COC_PROCESS_FOR_COC_TEAM.md:
```diff
@@ -0,0 +1,19 @@
+
+## Membership
+
+### Terms
+Members of the CoC Team will be added for a 1-year term and will be re-confirmed on
```
```suggestion
### Terms

Members of the CoC Team will be added for a 1-year term and will be re-confirmed on
```

VoltrexKeyva commented on conduct/COC_PROCESS_FOR_INCIDENT_MANAGEMENT.md:
```diff
@@ -1,7 +1,17 @@
+This document defines how the the OpenJS Foundation implements its code of conduct.
```
```suggestion
This document defines how the OpenJS Foundation implements its code of conduct.
```

eemeli commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,99 @@
+The OpenJSF makes the following commitments to reporters and victims of CoC incidents:
+
+1. All incident reports will be reviewed and investigated and will result in a response that is deemed necessary and appropriate to the circumstances.
+2. The identity of the victim(s) and reporter(s) of a code of conduct violation will be kept confidential.
```
Confidential from whom? Below it's mentioned that the CoC Team "is empowered to involve external people". Will the identities be kept confidential from them as well?

tobie replied:
I'm splitting those commitments up to make them more explicit. To answer your comment, however, external people are required to abide by the policy as stated, so that's covered.

eemeli replied:
Two separate thoughts here:
1. This means that someone reporting an incident cannot know ahead of time the set of people that their identity will be revealed to, on account of the CoC Team being empowered to bring in pretty much anyone. I guess that's ok?
2. There is no mention here of keeping the identity of someone accused of a violation confidential. Should that not be the case at least until the response has been determined?

tobie replied:
> This means that someone reporting an incident cannot know ahead of time the set of people that their identity will be revealed to, on account of the CoC Team being empowered to bring in pretty much anyone. I guess that's ok?

Yeah. Imagine there's an issue where the CoC Team needs to bring in a translator. There's no reasonable way to know their identity beforehand. It also doesn't matter. We're trying to avoid the case where someone reports a problem caused by someone and unknowingly report it to the very person that's causing the issue. And the current policy covers that imho.

> There is no mention here of keeping the identity of someone accused of a violation confidential. Should that not be the case at least until the response has been determined?

Yeah, that's a little more complicated. There are circumstances where that can't be the case (for example, if there's a risk of harm to additional people, legal considerations, etc.). Not sure how to properly manage this. Suggestions welcomed.

tobie replied:
I've added your second point as a task that needs to be resolved before merging this PR. Resolving this.

eemeli commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,99 @@
+
+The OpenJSF makes the following commitments to reporters and victims of CoC incidents:
+
+1. All incident reports will be reviewed and investigated and will result in a response that is deemed necessary and appropriate to the circumstances.
```
This seems to imply that all reports _will_ result in a response. Is it possible for a report to result in _no_ response, or would e.g. replying to the reporter that no action will be taken also considered a "result"?

tobie replied:
Yeah. "We won't take action" is a response.

eemeli commented on conduct/COC_PROCESS_FOR_INCIDENT_MANAGEMENT.md:
```diff
@@ -51,7 +53,7 @@ All members of the CPC are subscribed to the `report@openjsf.org` mailing list.
 Note: We understand that the reporter and the target can be separate persons. In this case the CPC will communicate with the reporter unless the target gives permission for the CPC to communicate with them.
 
-All members of the [Code of Conduct Panel (CoCP)](https://github.com/openjs-foundation/cross-project-council/blob/HEAD/conduct/FOUNDATION_CODE_OF_CONDUCT_REQUIREMENTS.md#code-of-conduct-panel)
+All members of the [Code of Conduct Panel (CoCP)](https://github.com/openjs-foundation/cross-project-council/blob/HEAD/conduct/COC_POLICY.md#code-of-conduct-panel)
```
Should there references here and below to the CoCP be updated?

tobie replied:
Thanks for catching that!

eemeli commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,99 @@
+
+Victim(s) and offender(s) of an incident have thirty (30) days to appeal to the CoC Team the decision made in response to an incident report if they believe the process was not properly followed.
+
+If the initial report was already managed by the CoC Team, a different set of CoC Team members should be chosen to manage the appeal process.
```
What is the minimum size for such a set? This seems to imply that the team needs to have at least twice that many members to be able to fulfill this step of the process.

tobie replied:
That's a process question around the makeup of the CoC Team that I'm purposely leaving out of the policy.

tobie replied:
The way I imagine this would be handled is by looping in more folks from the CPC and/or the Board in case there's a real issue.

tobie replied:
Also, I deliberately used _should_ here to account for situations where that would actually not be possible for lack of members.

tobie replied:
I've made this even more explicit now.

tobie commented:
Thank for your thorough review, @eemeli. I think I've answered or addressed the issues you raised. Please mark them as resolved or LMK if that's not the case. Thanks again for chiming in. Much appreciated!

Relequestual commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,107 @@
+
+Victim(s) and offender(s) of an incident have thirty (30) days to appeal to the CoC Team the decision made in response to an incident report if they believe the process was not properly followed.
+
+If the initial report was already managed by the CoC Team, a different set of CoC Team members should be chosen to manage the appeal process. The Coc Team may choose to involve external people if needed.
```
```suggestion
If the initial report was already managed by the CoC Team, a different set of CoC Team members should be chosen to manage the appeal process. The CoC Team may choose to involve external people if needed.
```

Relequestual commented on conduct/COC_PROCESS_FOR_INCIDENT_MANAGEMENT.md:
```diff
@@ -51,21 +53,22 @@ All members of the CPC are subscribed to the `report@openjsf.org` mailing list.
+        project leadership). If the leadership representatives expresses concerns then the contact person will bring these back to the CoC Team members for further discussion. This cycle can continue until the CoC Team members reach consensus that the leadership representatives concerns have been adequately addressed.
      1. Resolution shared: The resolution is shared with the reporter or target by the contact person. If the reporter or target expresses concerns then the contact person will bring these back to the CPC members for further discussion. This cycle can continue until the CPC members reach consensus that the reporter or target's concerns have been adquately addressed. If no feedback is received wit ...
-     1. Final resolution: the resolution agreed by the CoCP members is implemented and the outcome reported back to the reporter or target bythe main contact.
+     1. Final resolution: the resolution agreed by the CoC Team members is implemented and the outcome reported back to the reporter or target bythe main contact.
```
```suggestion
     1. Final resolution: the resolution agreed by the CoC Team members is implemented and the outcome reported back to the reporter or target by the main contact.
```

Relequestual replied:
Fix typo.

ctcpip commented on CODE_OF_CONDUCT.md:
```diff
@@ -105,13 +102,11 @@ representative at an online or offline event.
 
 All community leaders are obligated to respect the privacy and security of the
-reporter of any incident.
+reporter(s) and victim(s) of any incident.
```
this should include the (alleged) offender(s) as well

tobie replied:
Fixed the language. Added your initial point as a task to complete before merging this PR. Marking this as resolved.

ctcpip commented on CODE_OF_CONDUCT.md:
```diff
@@ -23,23 +14,29 @@ All recipients of reports commit to:
+
+### Appeal a decision
 
+The victim(s) and offender(s) have thirty (30) days to appeal the decision made in response to an incident report if they believe the process was not properly followed. Appeals are not suspensive.
```
shouldn't this be the reporter rather than "victim"?

and is a "victim" a necessary component of a CoC violation?  "victim" is a loaded word, and may not apply to all CoC violations

tobie replied:
Fixed the language issue.

> shouldn't this be the reporter rather than "victim"?

Unless the reporter is themself harmed, I don't think they should have the right to appeal. They're not a party in the process, imho.

ctcpip commented on CODE_OF_CONDUCT.md:
```diff
@@ -23,23 +14,29 @@ All recipients of reports commit to:
+
+### Appeal a decision
 
+The victim(s) and offender(s) have thirty (30) days to appeal the decision made in response to an incident report if they believe the process was not properly followed. Appeals are not suspensive.
```
agree.  I think it's better to refer to 'individuals' involved, and keep everything general to avoid characterizing anyone, especially erroneously, as victims, offenders, etc.

tobie replied:
Fixed this by adopting CNCF language here.

tobie commented:
@bensternthal and @ctcpip: good catch on the victim/offender language. [CNCF uses](https://github.com/cncf/foundation/blob/main/code-of-conduct/coc-incident-resolution-procedures.md) “person harmed” and “accused person.” Would that work for you?

bensternthal commented:
> @bensternthal and @ctcpip: good catch on the victim/offender language. [CNCF uses](https://github.com/cncf/foundation/blob/main/code-of-conduct/coc-incident-resolution-procedures.md) “person harmed” and “accused person.” Would that work for you?

Yes, we should use the same language as CNCF for these terms.

bnb commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,107 @@
+
+[CoC]: https://code-of-conduct.openjsf.org/
+[Charter]: https://github.com/openjs-foundation/cross-project-council/blob/main/CPC-CHARTER.md
+[CoC Team Charter]: https://github.com/openjs-foundation/cross-project-council/blob/main/conduct/COC_TEAM_CHARTER.md
```
This file does not exist

tobie replied:
🤦 Thanks for catching this.

Your broader point that this file needs to actually contain content to merge the PR is well taken and I agree.

tobie replied:
Resolving this as I've added a task to complete the CoC Team charter before merging this PR.

bnb reviewed (changes requested):
Please ensure the COC_TEAM_CHARTER file exists :)

tobie commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,107 @@
+
+# Code of Conduct Policy
+
+The OpenJS Foundation (OpenJSF) delegates to the Cross Project Council (CPC), via its [charter][], the responsibility of choosing, maintaining, and enforcing a Code of Conduct (CoC) for all of its projects, collab spaces, and for any related activities.
```
I cleaned up the language here to clarify the scope of the escalation and appeal process. I think they strike a good balance and give the project a fair amount of autonomy provided they run a tight ship.

Let me know your thoughts about the updated language.

ovflowd replied:
It's still a good move that this info gets relayed to each project TSC. For example, how does the Node.js Moderation team feel regarding the fact that the OpenJS CoC team can override decisions made by the Node.js CoC (Moderation) team?

Imagine the situation where the Node.js moderators decide that said user is not violating the CoC, but then the OpenJS CoC team overrides said decision. Not sure I'd be happy with that.

I thought that OpenJS CoC would only be called-in if each project's CoC/Moderation teams are unable to reach consensus, no?

ovflowd replied:
Right, would love to chatter about that on our next meeting.

tobie replied:
Sounds good. Let's keep this labeled as unresolved until then.

ovflowd replied:
As we discussed on last meeting, individual CPC representatives should already have done the relaying of information at this point in time.

+ For what it's laid here, the CPC CoC can only override decisions when there's a conflict of interest of emergencies as outlined in this PR.

Resolving this thread, but feel free to reopen if questions remain.

ovflowd commented on CODE_OF_CONDUCT.md:
```diff
@@ -1,17 +1,8 @@
+# OpenJS Foundation Code of Conduct
 
-The OpenJS Foundation and its member projects use [Contributor Covenant v2.0](https://contributor-covenant.org/version/2/0/code_of_conduct) as their code of conduct. The full text is included below in English, and [translations](https://www.contributor-covenant.org/translations) are available on the Contributor Covenant website.
+## Report an incident
```
IMO we should mention at the top where people can read our Code of Conduct.

tobie replied:
This document _is_ the code of conduct itself. Can you clarify your suggestion?

ovflowd replied:
The Code of Conduct itself is not mentioned here, just the guidelines of reporting incidents. The actual code of conduct that is in another document should actually be included or at least referenced here.

ovflowd replied:
Right, but it pretty much only explains the reporting pieces of our CoC, it doesn't mention the bylaws and the actual ettiquete and rules that the community should follow.

Hence, mentioning the remaining documents here, would make sense. We were using the Contributor Covenant, and we mentioned it here. Now we are moving towards an in-house baked guideline, but it's not mentioned anywhere within this document.

tobie replied:
We're still using the Contributor Covenant. CODE_OF_CONDUCT.md _is_ the code of conduct and _is_ Contributor Covenant. From the perspective of someone reporting an incident, little has changed.

ovflowd replied:
I don't see any mention of the Contributor Covenant anywhere. I think its fundamental that there are mentions of the actual code of conduct.

How will people know what are the rules they should abide by? I don't see a mention of this anywhere.

tobie replied:
The content of `CODE_OF_CONDUCT.md` is (and always has been) the Contributor Covenant. I've just removed the reference to if at the top to avoid getting people confused between the version that we are running and the version on the Contributor Covenant website, as we've agreed to partially modify it and it's now only 99.5% equivalent.

ovflowd replied:
That might be the source of my confusion here. Let's chat over the meeting today.

ovflowd commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,109 @@
+
+A common code of conduct allows a common escalation and appeal process and allows better support of projects in their efforts to apply the code of conduct.
+
+All OpenJSF projects, collab spaces, and any activities related to the OpenJSF are operating under this code of conduct unless otherwise stated. (For example, events organized in partnership with other organizations might be operating under a different code of conduct.)
```
How do we dispatch updates of our CoC to project's CoC's? For example, should Node.js's CoC simply redirect to OpenJS's CoC? (https://github.com/nodejs/admin/blob/HEAD/CODE_OF_CONDUCT.md)

tobie replied:
That's a great point. It seems related to https://github.com/openjs-foundation/cross-project-council/pull/889. Can we address it there, separately from this PR?

tobie replied:
We already changed the CoC on a number of occasions without a dedicated process, so I don't think we should block on this. Let's not forget that the CPC is composed of representatives of projects precisely so that decisions taken here reflect project needs and don't take anyone by surprise.

ovflowd replied:
I have mixed feelings here as for merging this halfway... with open requirements. But not strong enough to oppose to this approach.

My take is that knowing how slow certain pieces of our collaboration process are... We might at some point have an incomplete CoC process/charter for a while until all pieces get finally merged together.

Shouldn't we then merge this to a "feature-branch" and have it all merged to "main" after done?

tobie replied:
A dedicated process for changing the CoC is not a missing piece, it's an additional feature. The CoC is under the CPC's purview and thus, without any dedicated process for changing the CoC, regular CPC process applies.

ovflowd replied:
I see, that's fair and I was under the impression we were changing that. Thx for clarifying!

Sometimes large PR diffs make it confusing to know what's really going on.

ovflowd commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,109 @@
+
+All OpenJSF projects, collab spaces, and any activities related to the OpenJSF are operating under this code of conduct unless otherwise stated. (For example, events organized in partnership with other organizations might be operating under a different code of conduct.)
+
+Projects joining the foundation that already have a code of conduct will need to adopt the OpenJSF's CoC within three (3) months. Projects that don't already have a code of conduct operate under the OpenJSF's CoC as soon as they enter incubation.
```
Should this be mentioned here? This should be said on this document (https://github.com/openjs-foundation/cross-project-council/blob/main/PROJECT_PROGRESSION.md) instead of here. This sort of information feels only relevant for projects joining the Foundation.

tobie replied:
I understand that perspective and agree that this information needs to be repeated in the onboarding-related doc, but the decision to give some leniency to projects joining the foundation is a policy decision and belongs here, imho.

tobie replied:
Yes! Would you want to take a stab at it?

ovflowd replied:
wdym? do you want me to make the proposed changes? 👀

tobie replied:
Yeah, I was thinking mainly about updating the issue template [here](https://github.com/openjs-foundation/project-status/blob/main/.github/ISSUE_TEMPLATE/project-onboarding-checklist-template.md) alongside updating `PROJECT_PROGRESSION.md`.

ovflowd replied:
I can work on that, can we add a separate "tasklist" for this on the main CoC refactor issue? (If you don't mind doing that, I can then start working on a PR)

ovflowd replied:
_also feel free to resolve this specific thread once you've added the task on the issue and assigned it to me 🙇_

tobie replied:
See https://github.com/openjs-foundation/cross-project-council/issues/1255.

ovflowd commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,109 @@
+
+The name and contact details of the people responsible for managing the incident report should be provided to reporters as soon as possible (to uncover conflicts of interest upfront and allow for escalation).
+
+## Incident Response
```
Node.js over the years developed quite a good handbook for handling Moderation incidents (Moderation Policy): https://github.com/nodejs/admin/blob/main/Moderation-Policy.md

What about adopting it? It feels quite complete and very well written...

tobie replied:
@ovflowd: Could you please file a separate issue for your suggestion to adopt a uniform moderation approach across the foundation? This seems like a well-scoped and distinct enough issue that we can keep separated.

tobie replied:
@PaulaPaul: wrt to getting inspiration from the CNCF, my belief was that this was more about the CoC Team structure and its processes than about moderation. Looking forward to chat about it tomorrow, though.

ovflowd replied:
> Social media or other online behavior targeted at contributors in the context of their contributions to the Foundation?

As much as I'd love that, I wonder how we'd enforce CoC on platforms not governed by us?

ovflowd replied:
> @ovflowd: Could you please file a separate issue for your suggestion to adopt a uniform moderation approach across the foundation? This seems like a well-scoped and distinct enough issue that we can keep separated.

I may, but I don't think we can merge this PR as it is without actually providing these processes here; Since this PR effectively changes the moderation processes, and in a technical standpoint, it doesn't make sense to merge "tech debt"; I wouldn't merge this PR changes to moderation processes if we already have plans to update them in an uniform way. Wdyt?

tobie replied:
@PaulaPaul wrote:
> My thoughts there were not about enforcement but for the scope of the CoC. For example, if I am harassed online because of work I contributed to the foundation, that violation of the CoC would be in social media, not in a comment on a repo or in Slack. If that ever happened to me I would like to know if I could raise it as a violation of the CoC (one way or another). It's not realistic to think the CPC could police social media but could act as a place to report CoC violations of any kind specifically related to foundation contributions. Does that make sense?

This is mostly covered in the [scope section of the CoC](https://github.com/openjs-foundation/cross-project-council/blob/main/CODE_OF_CONDUCT.md#scope):

> _This Code of Conduct applies within all community spaces, and also applies when an individual is officially representing the community in public spaces. Examples of representing our community include using an official e-mail address, posting via an official social media account, or acting as an appointed representative at an online or offline event._

ovflowd replied:
> Are we talking about incident report management here, or about moderation? Moderation (i.e. pro-actively managing comments, etc., before incidents happen and are reported) is managed by the projects themselves. My understanding is that the policy you linked is a moderation policy, not a process to manage incident reports. 

Yes and No. The Moderation process/policy of Node.js describes an overall guidelines how Moderators should handle incidents (either proactively or through reports);

And I believe moderation in-repository is fundamental here, too; We do, after all (CPC and the Foundation) live within our Slack and this GitHub organization.

ovflowd replied:
I've created an Issue: https://github.com/openjs-foundation/cross-project-council/issues/1215

tobie replied:
Can we mark this as resolved? @PaulaPaul if you believe the scope of the CoC should be changed, can you file a separate issue for that please?

ovflowd reviewed (changes requested):
Unfortunately I don't feel that some of the changes here are clear enough and I don't feel comfortable with getting this merged without explicit reference of the actual code of conduct. The new structure feels confusing at best for me.

tobie commented:
@ovflowd wrote:
> Unfortunately I don't feel that some of the changes here are clear enough and I don't feel comfortable with getting this merged without explicit reference of the actual code of conduct. The new structure feels confusing at best for me.

Hey Claudio, let's chat about it in the working session today. I think you're misunderstanding the PR. The content of CODE_OF_CONDUCT.md is and always has been the Contributor Covenant. I've removed the reference the Contributor Covenant at the top because we agreed to change a few small things in it and I wanted to avoid having people confused over which one was the source of truth. We're not introducing any significant changes to the CoC here, the core of the changes you're seeing in this PR are the implementation details being spelled out.

ovflowd commented:
> @ovflowd wrote:
> 
> > Unfortunately I don't feel that some of the changes here are clear enough and I don't feel comfortable with getting this merged without explicit reference of the actual code of conduct. The new structure feels confusing at best for me.
> 
> 
> 
> Hey Claudio, let's chat about it in the working session today. I think you're misunderstanding the PR. The content of CODE_OF_CONDUCT.md is and always has been the Contributor Covenant. I've removed the reference the Contributor Covenant at the top because we agreed to change a few small things in it and I wanted to avoid having people confused over which one was the source of truth. We're not introducing any significant changes to the CoC here, the core of the changes you're seeing in this PR are the implementation details being spelled out.

Thanks! Yeah, I'd super appreciate figuring out that during the meeting, I bet Im just confused over a few silly details.

tobie commented:
@ovflowd: essentially, the content of `CODE_OF_CONDUCT.md` from line 48 onwards is the same content as the whole of the Contributor Covenant:

- `CODE_OF_CONDUCT.md`: https://github.com/openjs-foundation/cross-project-council/blob/main/CODE_OF_CONDUCT.md?plain=1#L48-L179
- Contributor Covenant 2.1: https://www.contributor-covenant.org/version/2/1/code_of_conduct/code_of_conduct.md

ovflowd commented:
> @ovflowd: essentially, the content of `CODE_OF_CONDUCT.md` from line 48 onwards is the same content as the whole of the Contributor Covenant:
> 
> 
> 
> - `CODE_OF_CONDUCT.md`: https://github.com/openjs-foundation/cross-project-council/blob/main/CODE_OF_CONDUCT.md?plain=1#L48-L179
> 
> - Contributor Covenant 2.1: https://www.contributor-covenant.org/version/2/1/code_of_conduct/code_of_conduct.md

That looks alright. Thanks for explaining this to me!

ovflowd reviewed (approved):
After yesterday's CPC meeting it became clear, at least for the members attending the meeting that this has matured enough to the point we should proceed with it.

Fellow, CPC members I ask y'all on your earliest convenience to review/approve this PR. 

cc @openjs-foundation/cpc

ljharb commented on CODE_OF_CONDUCT.md:
```diff
@@ -23,23 +14,29 @@ All recipients of reports commit to:
+
+### Appeal a decision
 
+Harmed individuals and accused individuals may [appeal](#appeals) the decision to the CoC Team up to thirty (30) days after a response was provided if they believe that the process was not followed properly. Appeals are not suspensive.
```
i have no idea what "appeals are not suspensive" means; i've never heard this. Can this be rephrased?

mhdawson replied:
I believe it would mean that if you appeal is does not immediately affect what you are appealing. However, I agree wording it differently would make it more understandable.

tobie replied:
Would: "Appeals do not delay or block the execution of decisions made in the response." Work for you?

tobie replied:
```suggestion
Harmed individuals and accused individuals may [appeal](#appeals) the decision to the CoC Team up to thirty (30) days after a response was provided if they believe that the process was not followed properly. Appeals do not delay or block the execution of decisions communicated in the response.
```

ljharb commented on CODE_OF_CONDUCT.md:
```diff
@@ -23,23 +14,29 @@ All recipients of reports commit to:
+Harmed individuals and accused individuals may [appeal](#appeals) the decision to the CoC Team up to thirty (30) days after a response was provided if they believe that the process was not followed properly. Appeals are not suspensive.
 
-## Report an issue in a space managed by the foundation
+To [appeal][] a decision, please email the [Code of Conduct Team (CoC Team)][CoC Team] at <coc-appeal@lists.openjsf.org>. TODO: create this address
```
aside: it seems weird to me to have `lists.openjsf.org` rather than just `openjsf.org` on emails

tobie replied:
Agreed. @bensternthal, would that be possible?

bensternthal replied:
I created coc@openjsf.org which forwards to  report@lists.openjsf.org

ljharb commented on CODE_OF_CONDUCT.md:
```diff
@@ -105,13 +102,11 @@ representative at an online or offline event.
 
 All community leaders are obligated to respect the privacy and security of the
-reporter of any incident.
+harmed individuals and the reporter(s) of any incident.
```
after this is landed i'll remove all the hard wraps, consistent with #1116.

tobie replied:
Thank you!

ljharb commented on conduct/COC_TEAM_CHARTER.md:
```diff
@@ -0,0 +1,92 @@
+
+### Member selection
+
+The CPC will hold an annual selection process in order to select primary and alternate community members.
```
when will this process be held?

tobie replied:
As soon as we ship this!

mhdawson reviewed (approved):
LGTM

bnb commented on conduct/COC_TEAM_CHARTER.md:
```diff
@@ -0,0 +1,92 @@
+- Handle code of conduct [appeals][appeal].
+- Keep the CPC up to date of cases requiring the CPC's attention.
+- Provide regular transparency reports to the CPC.
+- Define, document, and maintain all processes necessary to its operations, notably the [CoC Process for Incident Management](COC_PROCESS_FOR_INCIDENT_MANAGEMENT.md).
```
IMO self-definition is not the correct choice here, and I'm concerned that it's on the table. Any form of moderation team _should_ be people enabled to act on well-documented and well-defined processes and rules. To remove the potential for bias, **they should _not_ be charged with writing those rules themselves.**

In our case, writing these rules, documenting these rules, and maintaining the processes around those rules should be the charge of the CPC, not the CoC Team.

tobie replied:
That's the role of policy. The standard practice of good governance is to separate policy from process and offer a system to appeal the process if there's ground to believe it isn't correctly implementing policy. Moving process implementation to the CPC is the equivalent of asking legislators to author technical standards or shifting security right instead of left.

bnb replied:
Commented again on this here https://github.com/openjs-foundation/cross-project-council/pull/1135#pullrequestreview-1839768804 but I strongly and fundamentally disagree with this. I understand the concept you're operating off of but fundamentally I disagree with it and think that for community safety issues, it's an approach that opens up the opportunity for issues from both ends - it creates potential for abuse, and it creates the opportunity for decisions to be questioned.

PaulaPaul replied:
Repeating this comment here on suggestion from @tobie -

From the content proposed in this branch, there is still some work to do on [this particular file](https://github.com/openjs-foundation/cross-project-council/blob/coc/conduct/COC_TEAM_CHARTER.md) but the groundwork is there for eligibility. For this item, I think we could come to closure more efficiently by landing this PR and opening a specific issue (and tracking it on the CPC calls).

It does appear that the CoC Team composition and operations is a specific area that requires additional work. I propose we open this as a separate and specific issue so as not to block the entire CoC policy. That issue can be highlighted on our regular CPC calls-

tobie replied:
This was fixed. It is now the CPC which is in charge of this. Tracked in #1256.

bnb commented on conduct/COC_TEAM_CHARTER.md:
```diff
@@ -0,0 +1,92 @@
+
+Members of the CoC Team serve a one year term. There are no limits on the number of terms a CoC Team member may serve.
+
+### Chair
+
+The primary members will select a chair from one of the three primary community members. When the chair isn't involved in a case, the members handling the case must chose a chair among themselves.
+
+The chair is responsible for handling the case according to documented processes, and making sure that the CoC Team upholds this charter, the [code of conduct][CoC] and the [code of conduct policy][CoC Policy].
```
I don't think this is necessary and simply adds more process.

tobie replied:
What does? Having a chair?

tobie replied:
Removed CoC Chair in https://github.com/openjs-foundation/cross-project-council/pull/1135/commits/9e71b96d7375ff8813624946d7cbe1b6c7d7a2e9.

bnb commented on conduct/COC_TEAM_CHARTER.md:
```diff
@@ -0,0 +1,92 @@
+- CPC member can block a candidate if they can motivate the reason why
+]
+
+### Eligibility
+
+Primary and alternate members of the CoC Team must: 
+
+* Commit to uphold the Foundation's [Code of Conduct][CoC]
+* Commit to uphold the Foundation's [Code of Conduct Policy][CoC Policy]
+* Commit to uphold this charter
+* Be willing to treat every accused individuals in the same way regardless of the impact of their contributions to open source
```
I don't know if this is worth including, as it seems more ideological rather than substantial. It's the only section of this document that is written that way.

For clarity, what I'd consider substantial in this section are things like "cannot exceed 20% representation from the same employer".

tobie replied:
Following our conversation on Friday Jan 26, we agreed to keep the criteria mentioned above. I've also included an additional provision to limit the number of primary members with the same employer in https://github.com/openjs-foundation/cross-project-council/pull/1135/commits/7f817b4aa479dfaa00bdf2b9e42e5fc1e944c53a.

@bnb please add any other criteria you had in mind and resolve the conversation when you're good.

bnb commented on conduct/COC_TEAM_CHARTER.md:
```diff
@@ -0,0 +1,92 @@
+
+The CoC Team is composed of five (5) primary members and a variable number of alternate community members.
+
+### Primary members
```
Alternate membership is never defined. Based on how this team charter is written, this needs to be well-defined. I am blocking on this.

tobie replied:
Both primary and alternate members are defined in the charter, in terms of their roles. Could you clarify how else you'd want them defined?

bnb replied:
Please link where both primary and alternate members are defined? This file is called "COC_TEAM_CHARTER.md", so I assume it's the charter you're talking about. Searching throughout the full PR contents, I'm not able to find definitions - I'm bad at Control + F sometimes, though!

The section I left this comment on is titled "Primary members" but there is no "Alternate members" section - looking for the definition of "Alternate member" I realized that Primary members actually _also_ aren't defined here, which it probably should be.

tobie replied:
Fixed.

tobie replied:
@bnb can you please check this fix and resolve this conversation if you're happy with the fix?

tobie commented on conduct/COC_PROCESS_FOR_INCIDENT_MANAGEMENT.md:
```diff
@@ -27,10 +33,6 @@ Personal information is confidential. All reports should be recorded, together w
 
 For each report there should be an issue which captures the report, discussion and final conclusion. The report should NOT contain identifying details of the reporter.
 
-## Conflict of Interest
-
-Any member of the CPC or Code of Conduct Panel(CoCP) who is involved in the report should recuse themselves from the discussions.
-
```
Yes. It's policy, not process (and so has been moved to the policy doc).

tobie commented on conduct/COC_PROCESS_FOR_INCIDENT_MANAGEMENT.md:
```diff
@@ -51,21 +53,22 @@ All members of the CPC are subscribed to the `report@openjsf.org` mailing list.
-  * If no response has been sent within 1 day the Foundation Executive Director will ensure a member of the Code of Conduct
-    Panel is identified as the main point of contact either through email or scheduling a meeting for the CoCP.
+  * One of the Code of Conduct Team (CoC Team) members will respond to the reporter or target confirming that the report has been received. When possible the same person who responds initially will act as the point of contact for future conversations.
+  * If no response has been sent within 1 day the Foundation Executive Director will ensure a member of the CoC Team is identified as the main point of contact either through email or scheduling a meeting for the CoC Team.
```
Agreed. As [mentioned above](https://github.com/openjs-foundation/cross-project-council/pull/1135/files/cb915764bb11519cb71ba36d649382d623172411#diff-7cdf4971fc5d0050c658f571b2309da3ca21c1b9adefe2d86b3b42fd4849ec0fR9), these haven't been touch and will need to be updated by the CoC Team.

bnb replied:
Commented on that here 😅 https://github.com/openjs-foundation/cross-project-council/pull/1135#pullrequestreview-1839768804

tobie replied:
Addressed this in #1256.

tobie commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,120 @@
+
+All OpenJSF projects, collab spaces, and any activities related to the OpenJSF are operating under this code of conduct unless otherwise stated. (For example, events organized in partnership with other organizations might be operating under a different code of conduct.)
+
+Projects joining the foundation that already have a code of conduct will need to adopt the OpenJSF's CoC within three (3) months. Projects that don't already have a code of conduct operate under the OpenJSF's CoC as soon as they enter incubation.
```
I see how they make the whole thing look overly formal. However, I find they make the content more accessible (e.g. I find them helpful). Fine either way, though.

tobie commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,120 @@
+
+## Conflict of Interest
+
+Any person involved with the management of an incident report must immediately notify the CoC Team in writing and recuse themselves from the process if they have been involved in the incident itself or have any other conflict of interest with any of the involved parties (e.g. same employer).
```
Agree we should have a separate conflict of interest policy. Don't think we should block on this, though, as it's not like we had one before.

bnb replied:
> Don't think we should block on this, though, as it's not like we had one before.

Eh, at this point I'm worried about this _never_ getting done if we don't do it now. Knowing how we work, I'd prefer if we solved it now before merging.

tobie replied:
Agreed to carve out clearer definition of conflict of interest in followup PR. Tracking in #1238.

tobie replied:
@bnb can you please confirm you're OK with deferring this by resolving this conversation? Thanks!

tobie commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,120 @@
+
+The CoC Team is empowered to manage escalation requests of incident reports managed by projects that have opted-in to enforce the code of conduct themselves.
+
+If the CoC Team finds that the escalation request has merit, the CoC Team may decide to takeover the management of the incident report or request changes to the existing management process. 
```
I think it's important for projects to be accountable to properly implementing the policies agreed to by project representatives in the CPC.

bnb replied:
I think that is orthogonal to my point, and is an extremely blunt way to manage this situation that removes nuance that will _always_ exist when something like this comes up. If this ever happens, that as the answer will not suffice.

PaulaPaul replied:
Copying this comment here (breaking out individual comments as requested by @tobie)

Since there isn't consensus on the item above (or, how to reach consensus) is it possible to leave this as is, but then make this a specific part of the follow up activity needed to review CoC updates with each individual project (allowing us to land the PR but expecting we will need to open additional issues specific to the escalation and appeal policy?)

An option here is to open an issue with each project to specifically and openly request approval and also openly document which projects opt-in to managing their own CoC reporting.

tobie replied:
> An option here is to open an issue with each project to specifically and openly request approval

The CPC _is_ the forum for working this approval out.

> and also openly document which projects opt-in to managing their own CoC reporting.

That's already covered and not subject to disagreement.

tobie replied:
Fixed in 08811e8. @bnb mind looking at the language and seeing what you think of it? Thanks.

tobie commented on conduct/COC_TEAM_CHARTER.md:
```diff
@@ -0,0 +1,92 @@
+
+## Responsibilities
+
+[TODO: add links]
```
Thanks! There's a few left I still need to get to.

tobie reviewed (commented):
Some initial comments. Will snooze this until next week. Apologies in advance for the terseness.

tobie commented:
Thanks for the thorough review, @bnb!

PaulaPaul reviewed (approved):
I've reviewed these changes and comments - we have one blocking comment from @bnb regarding clear identification of conflict of interest and I believe @tobie has responded to that comment (comment link below)
https://github.com/openjs-foundation/cross-project-council/pull/1135#discussion_r1446407933

@bnb if you are not in agreement can you suggest additional text we could add to clarify?

bnb commented on conduct/COC_PROCESS_FOR_INCIDENT_MANAGEMENT.md:
```diff
@@ -1,7 +1,17 @@
+
+The [Code of Conduct Team (CoC Team)](https://github.com/openjs-foundation/cross-project-council/blob/HEAD/conduct/COC_POLICY.md#code-of-conduct-team-coc-team) is responsible for the maintenance of these processes.
+
+**These processes haven't been updated since the creation of the CoC Team. It is expected that the CoC Team will overhaul them in the near future.**
```
Extremely large -1 against this. The CoC team should not be self-governing, and setting it up as such is extremely dangerous.

bnb replied:
More specifically, the CoC team should be _implementing_ rules and structures that the CPC sets up for them, asking for clarification on interpretations as needed. This way, there is no question about whether or not the actions of the CoC team to be questioned in the decisions they've made - they are just enacting what the CPC set up. It's shielding for both the CPC and the team.

Self-governance is something that's _actively_ avoided by the Node.js Moderation Team for the above reason. Also we've seen at least one other Foundation that's had pretty explosive issues with self-governance - justified or not.

tobie replied:
Would setting up a ratifying process for the CPC alleviate your concerns?

PaulaPaul replied:
Repeating this comment here on suggestion from @tobie -

From the content proposed in this branch, there is still some work to do on [this particular file](https://github.com/openjs-foundation/cross-project-council/blob/coc/conduct/COC_TEAM_CHARTER.md) but the groundwork is there for eligibility. For this item, I think we could come to closure more efficiently by landing this PR and opening a specific issue (and tracking it on the CPC calls).

It does appear that the CoC Team composition and operations is a specific area that requires additional work. I propose we open this as a separate and specific issue so as not to block the entire CoC policy. That issue can be highlighted on our regular CPC calls-

tobie replied:
Fixed in d6010efa4394e309d9ea68510e639f4371965f0d by leaving the responsibility of defining and maintaining the enforcement processes with the CPC. Of course, this doesn't mean that the CoC Team members cannot contribute to those, just that they remain under the responsibility of the CPC.

tobie replied:
@bnb can you please check this fix and resolve this conversation if you're happy with the fix?

wesleytodd commented on CODE_OF_CONDUCT.md:
```diff
@@ -23,23 +14,29 @@ All recipients of reports commit to:
     * **Webhint:** <support@webhint.io>
-2. For every other OpenJS Foundation project, please email <report@lists.openjsf.org>. The Cross Project Council (CPC) is responsible for managing these reports.
+    
+    In case of a conflict of interest or if the report is not handled appropriately, you may [escalate the report][escalation] to the [Code of Conduct Team (CoC Team)][CoC Team] by emailing <coc-escalation@lists.openjsf.org> or by contacting any of its members.
```
> What constitutes a conflict of interest?

Do we have some examples of wording from other CoCs? I am trying to think how I would write something on this, and it is hard to figure out what *isnt* a conflict in cases where we are all here because of our participation in projects or companies with specific interests. Thinking of the discussions in node right now, if "conflict of interest" included "being involved in any package manager project" then all the important voices would be excluded from any CoC violation discussion. Maybe that is good, but I agree getting this right is important and it would be good to have prior art to reference.

PaulaPaul replied:
On the topic of reports not being handled properly, breaking this comment out here as well:

The current text reads:
"In case of a conflict of interest or if the report is not handled appropriately...", so conflict of interest is part of mishandling, and perhaps the wording can be softened to say that if any parties involved believe the report has been mishandled in any way, for example, unnecessary delay or lack of communication..."

tobie replied:
> I think it is possible to define the term 'conflict of interest' and not tie a general Foundation policy.

Absolutely. If there's a simple conflict of interest policy everyone's happy with, it makes sense baking it in right away. However, given we've lived without a formal conflict of interest policy so far, we shouldn't block this PR on it if there isn't consensus about the policy's content. I'm opening a new issue for the conflict of interest policy. If we have consensus quickly, we can pull it in right away.

tobie replied:
> "In case of a conflict of interest or if the report is not handled appropriately...", so conflict of interest is part of mishandling, and perhaps the wording can be softened to say that if any parties involved believe the report has been mishandled in any way, for example, unnecessary delay or lack of communication..."

My understanding it that the concern isn't that people would hesitate to escalate, it's that the process might be weaponized against maintainers, particularly underrepresented maintainers, as we've seen happen elsewhere.

tobie replied:
Addressed in https://github.com/openjs-foundation/cross-project-council/pull/1135/commits/93b246450818722ad179c528b928d30db3eac9c9. @bnb can you please check if the the proposed fix addresses your request for exemplifying mismanagement and resolve this conversation if so. Thanks!

tobie commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,120 @@
+
+If the appeal follows an incident report managed by a project, the project is expected to participate in good faith in the appeal process and to share all the information related to the incident report with the CoC Team.
+
+Appeals are not suspensive.
```
```suggestion
Appeals do not delay or block the execution of decisions communicated in the response.
```

PaulaPaul commented:
Adding some thoughts below on open concerns-

> * [ ]   Address concerns raised by @bnb:
>   
>   * [ ]  There's no clear definition of conflict of interest. Agreement that we need a conflict of interest policy, but can we land this PR without it?

I think it is possible to define the term 'conflict of interest' and not tie a general Foundation policy to that definition, since the term here is specific to CoC issues. For that purpose, we could include a standard definition (there are many that are used by non-profit organizations and others). I thought this was a [decent example](https://www.compasspoint.org/sites/default/files/documents/conflictspolicy.pdf) (adopted from this [article](https://blueavocado.org/leadership-and-management/nonprofit-conflict-of-interest-a-3-dimensional-view/), which has other examples):

Definition: Conflict of Interest (with regard to CoC issues)
"A conflict of interest exists when the loyalties or actions of a CoC Team member are divided
 between the interests of the Foundation and the interest of the directors, officers, or community
 members who are subjects of a CoC issue or complaint.  Both the fact and the appearance of a
 conflict of interest should be avoided." 

The Policy, for purposes of CoC issues could also be adapted from that document (example below)
II. Policy
Directors, CoC Team or community members are expected to use good judgment, 
to adhere to high ethical standards, and to conduct their affairs in such a manner
as to avoid any actual or potential conflict between the personal interests of a 
director, CoC Team or community member and  and those of the Foundation.

>   * [ ]  What constitutes report mishandling needs to be defined more clearly.

The current text reads: 
"In case of a conflict of interest or if the report is not handled appropriately...", so conflict of interest is part of mishandling, and perhaps the wording can be softened to say that if any parties involved believe the report has been mishandled in any way, for example, unnecessary delay or lack of communication..."

>   * [ ]  The escalation and appeal policy described in this PR allow the CoC Team to reverse decisions made at the project level in certain fairly scoped situations. There isn't consensus about this.
>   * [ ]  The current policy delegates defining case handling processes to the CoC Team. There isn't consensus about this.
>   * [ ]  There isn't consensus that having a CoC Chair is useful.

Since there isn't consensus on the three items above (or, how to reach consensus) is it possible to leave this as is, but then make this a specific part of the follow up activity needed to review CoC updates with each individual project (allowing us to land the PR but expecting we will need to open additional issues specific to the escalation and appeal policy?)
An option here is to open an issue with each project to specifically and openly request approval and also openly document which projects opt-in to managing their own CoC reporting.

>   * [ ]  There was a comment about eligibility criteria being moot and not mentioning key issues like caping the number of CoC members with the same employer. Not sure where it went.

From the content proposed in this branch, there is still some work to do on [this particular file](https://github.com/openjs-foundation/cross-project-council/blob/coc/conduct/COC_TEAM_CHARTER.md) but the groundwork is there for eligibility. For this item, I think we could come to closure more efficiently by landing this PR and opening a specific issue (and tracking it on the CPC calls). 

@tobie @bnb happy to join a call and find a way to move this forward without losing the open questions/concerns-

tobie commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,126 @@
+
+If the CoC Team finds that the escalation request has merit, the CoC Team may decide to takeover the management of the incident report or refer the issue to the Cross Project Council which may request changes to the existing management process. 
+
+Projects are expected to participate in good faith in the escalation process and to share all the information related to the incident report with the CoC Team. The CoC Team is expected to consider project and community needs during the escalation process and may involve the Cross Project Council if necessary and project leadership if appropriate.
```
This is specific to appeals and escalation.

bnb commented:
Removed my rejection. I don't quite want to approve, but I've removed my blocking - this is close enough and I believe my major concerns have been addressed.

tobie commented:
This is now ready to be merged and will be merged by EOW (unless someone blocks).

ljharb commented on conduct/COC_POLICY.md:
```diff
@@ -59,7 +59,7 @@ A record of all communications related to an incident report, escalation request
 The identity of the harmed individual(s) and the and reporter(s) of an incident report must be kept confidential.
 
-The identity of the accused individual should be kept confidential. However, incidents requiring immediate action (e.g. to prevent further harm) and implementing the decision resulting from the incident management process might require sharing the identity of the accused individual beyond the people involved in the incident management process itself. In such cases, care should be taken to not mak ...
+The identity of the accused individual should be kept confidential. However, incidents requiring immediate action (e.g. to prevent further harm) and implementing the decision resulting from the incident management process might require sharing the identity of the accused individual beyond the people involved in the incident management process itself. In such cases, care should be taken to not mak ...
```
hmm, this feels a bit "react patents clause" punitive - it's more that, any confidential information publicly disclosed stops requiring confidentiality, but any non-disclosed information continues to require it.

tobie replied:
I think that the argument here is should there be partial or incorrect information made public that needs to be addressed or corrected. This might require providing more context.

tobie replied:
(I hear your point, though.)

ljharb replied:
The case where what’s made public is inaccurate/misleading is a tricky one, indeed, but in particular i think it’s important that the individual be able to reveal some parts of the confidential info while keeping other parts private.

tobie replied:
@ljharb fyi: I've reframed this a bit to try and better address your concern.

ljharb replied:
awesome, i think this is good enough to go forward :-) it'll likely need refining in the future but the intention seems captured to me.

eemeli commented on CODE_OF_CONDUCT.md:
```diff
@@ -23,23 +14,29 @@ All recipients of reports commit to:
+    In case of a conflict of interest, if the report is not handled appropriately (for example, if there are unacceptable delays, a lack of communication, confidentiality isn't respected, etc.), or in case of an emergency, you may [escalate the report][escalation] to the [Code of Conduct Team (CoC Team)][CoC Team] by emailing <coc@openjsf.org> or by contacting any of its members.
+2.  **To report any other incident** (for example: in an OpenJS Foundation project not listed above, in a Collab Space, in the Cross Project Council, or during an event), please email the [Code of Conduct Team (CoC Team)][CoC Team] at <coc@openjsf.org> or contact any of its members.
+
+### Appeal a decision
```
This section is weird, as it's mostly summarising content from `conduct/COC_POLICY.md#appeals`, but it's also defining from when the 30 days are counted, which isn't explicitly defined there.

Could this section be dropped, or replaced with a link to the policy's Appeals section? That would help deduplicate the text.

Also, the first "appeal" link doesn't work.

tobie replied:
Duplication is by design here, to offer a summary, in the CoC itself, for the benefit of reporters of their main course of action.

I think your below suggested fix addresses your other concern here.

eemeli commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,128 @@
+
+The CoC Team is empowered to manage escalation requests of incident reports managed by projects that have opted-in to enforce the code of conduct themselves.
+
+If the CoC Team finds that the escalation request has merit, the CoC Team may decide to takeover the management of the incident report or refer the issue to the Cross Project Council which may request changes to the existing management process. 
```
```suggestion
If the CoC Team finds that the escalation request has merit, the CoC Team may decide to take over the management of the incident report or refer the issue to the Cross Project Council which may request changes to the existing management process. 
```

eemeli commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,128 @@
+
+## Appeals
+
+Harmed individuals and accused individuals have thirty (30) days to appeal to the CoC Team a decision made in response to an incident report. Decisions may only be appealled on the grounds that the documented process was not properly followed or that it did not meet the requirements of this policy. If the CoC Team finds that the appeal has merit, it may decide to override the decision that was pr ...
```
```suggestion
Harmed individuals and accused individuals have thirty (30) days after a response was provided to appeal to the CoC Team a decision made in response to an incident report. Decisions may only be appealed on the grounds that the documented process was not properly followed or that it did not meet the requirements of this policy. If the CoC Team finds that the appeal has merit, it may decide to override the decision that was previously made.
```

eemeli commented on conduct/COC_TEAM_CHARTER.md:
```diff
@@ -0,0 +1,88 @@
+* Commit to uphold this charter
+* Be willing to treat every accused individuals in the same way regardless of the impact of their contributions to open source
+
+No two primary members can work for the same employer.
```
If this is the case, how can the current/initial CoC team include both the OpenJSF Executive Director and the OpenJSF Communications Manager?

ljharb commented on conduct/COC_TEAM_CHARTER.md:
```diff
@@ -0,0 +1,88 @@
+
+### Member selection
+
+The Cross Project Council (CPC) will hold an annual selection process in order to select primary and alternate community members.
```
presumably we'll hold this process (for the first time) ASAP once this PR lands?

VoltrexKeyva commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,128 @@
+
+The identity of the harmed individual(s) and the and reporter(s) of an incident report must be kept confidential.
+
+The identity of the accused individual should be kept confidential. However, incidents requiring immediate action (e.g. to prevent further harm) and implementing the decision resulting from the incident management process might require sharing the identity of the accused individual beyond the people involved in the incident management process itself. Additionally, the obligation of confidentialit ...
```
```suggestion
The identity of the accused individual should be kept confidential. However, incidents requiring immediate action (e.g. to prevent further harm) and implementing the decision resulting from the incident management process might require sharing the identity of the accused individual beyond the people involved in the incident management process itself. Additionally, the obligation of confidentiality towards the accused individual ends if the accused individual chooses to disclose the information publicly. In all cases, however, care should be taken to limit information sharing to what is strictly necessary to address the issue at hand and avoid making that sharing retaliatory.
```

VoltrexKeyva commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,128 @@
+
+### Incident management processes
+
+The Cross Project Council (CPC) defines, documents, and maintains all processes necessary for the management of incident reports and the enforcement of the code of code of conduct and of this policy through the [CoC Process for Incident Management](COC_PROCESS_FOR_INCIDENT_MANAGEMENT.md).
```
```suggestion
The Cross Project Council (CPC) defines, documents, and maintains all processes necessary for the management of incident reports and the enforcement of the code of conduct and this policy through the [CoC Process for Incident Management](COC_PROCESS_FOR_INCIDENT_MANAGEMENT.md).
```

VoltrexKeyva commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,128 @@
+
+The Cross Project Council (CPC) defines, documents, and maintains all processes necessary for the management of incident reports and the enforcement of the code of code of conduct and of this policy through the [CoC Process for Incident Management](COC_PROCESS_FOR_INCIDENT_MANAGEMENT.md).
+
+The CPC works with the OpenJS Foundation staff to develop infrastructure and processes necessary to comply to the requirements of this policy, notably when it comes to record keeping, confidentiality, and transparency.
```
```suggestion
The CPC works with the OpenJS Foundation staff to develop the infrastructure and processes necessary to comply with the requirements of this policy, notably when it comes to record-keeping, confidentiality, and transparency.
```

VoltrexKeyva commented on conduct/COC_POLICY.md:
```diff
@@ -0,0 +1,128 @@
+
+1. Commit to uphold the code of conduct.
+2. Commit to uphold this policy.
+3. Provide a general email address that incidents can be reported to and document it alongside the code of conduct.
```
```suggestion
3. Provide a general email address where incidents can be reported and document it alongside the code of conduct.
```

VoltrexKeyva commented on conduct/COC_TEAM_CHARTER.md:
```diff
@@ -0,0 +1,88 @@
+- Handle code of conduct [incident reports][incident report].
+- Handle code of conduct [escalation requests][escalation].
+- Handle code of conduct [appeals][appeal].
+- Keep the CPC up to date of cases requiring the CPC's attention.
```
```suggestion
- Keep the CPC up to date on cases requiring the CPC's attention.
```

VoltrexKeyva commented on conduct/COC_TEAM_CHARTER.md:
```diff
@@ -0,0 +1,88 @@
+
+At least two (2) of the members involved in handling a case must be community members and at least one (1) must be an OpenJS Foundation staff member.
+
+If not enough primary community members are available to meet those requirements, primary members should involve one or more alternate community members. Similarily, if no OpenJSF staff primary member is available, another OpenJSF staff representative, choosen by the staff, should be involved instead.
```
```suggestion
If not enough primary community members are available to meet those requirements, primary members should involve one or more alternate community members. Similarly, if no OpenJSF staff primary member is available, another OpenJSF staff representative, chosen by the staff, should be involved instead.
```

ljharb commented:
Seems like this PR removed this page: https://github.com/openjs-foundation/cross-project-council/blob/0c30a269acb2cc0b22cf3c62b8389e779bda0101~1/conduct/FOUNDATION_CODE_OF_CONDUCT_REQUIREMENTS.md but the link still exists somewhere, so the link checker is failing

tobie commented:
Must have missed this one which is in meeting minutes: https://github.com/openjs-foundation/cross-project-council/blob/2070ca3be62a291e5a0a3023e6233dd845119422/meetings/2019/2019-04-01.md?plain=1#L32

It's kind of weird to be editing minutes years later for this, tbh.

Holmes-Hitler commented:
Hello world

dawitalemayehutadese21-ops commented:
Everything view

dawitalemayehutadese21-ops commented on CODE_OF_CONDUCT.md:
```diff
@@ -1,17 +1,8 @@
-
-1. To report an issue in one of the projects listed below, please use the method provided. The project itself is responsible for managing these reports.
+1.  To report an incident in one of the projects listed below, please use the method provided. The project itself is responsible for managing these reports.
     * **AMP Project:** <code-of-conduct@amp.dev>
```
[](url)
